pages/api/search.ts

```diff
@@ -0,0 +1,135 @@
+import { Client as ESClient } from '@elastic/elasticsearch';
+import { allAuthors, allBlogs, Authors } from 'contentlayer/generated';
+
+const idToPost = (id: string) => allBlogs.find((p) => p.slug === id) || null;
+
+const authorFromHandle = (handle: string) => allAuthors.find((a) => a.slug === handle) || null;
+
+const client = new ESClient({
+  node: {
+    url: new URL(process.env.ELASTIC_URL),
+  },
+  auth: {
+    username: process.env.ELASTIC_USERNAME,
+    password: process.env.ELASTIC_PASSWORD,
+  },
+  maxRetries: 5,
+  requestTimeout: 60000,
+});
+
+export interface AuthorView {
+  slug: string;
+  avatar: string;
+  name: string;
+}
+
+const authorToView = (author: Authors) => ({
+  slug: author.slug,
+  avatar: author.avatar,
+  name: author.name,
+});
+
+export interface SearchResult {
+  id: string;
+  title: string;
+  description: string;
+  permalink: string;
+  score: number;
+  tags: string[];
+  summary: string;
+  date: string;
+  authors: AuthorView[];
+}
+
+export interface SearchResponse {
+  status: string;
+  error?: string;
+  query: string;
+  data: SearchResult[];
+}
+
+export default async function handler(req, resp) {
```

Comment:
I would recommend the search from https://github.com/tailscale-dev/tailscale-dev/pull/269#issue-1810851839